### PR TITLE
Restructured Language Server Component

### PR DESCRIPTION
**Pull request by a-sr**
- Adapted to use Maven artifacts of KLighD
- Moved LS code that is no longer a plugin
- Bumped Eclipse versions to latest release

**TODO** => @NiklasRentzCAU or @soerendomroes :
- [x] Test language server in Eclipse
- [x] Test language server artifact in VS Code extension
- [x] Check other development streams (Elk, Klighd) 
  - [x] Adopt new target platform

### CODE REVIEW

**Comment by soerendomroes** [comment]
I tested compilation and simulation of Abro as well as several synthesis options for the language server artifact in the language server extension.

**Comment by soerendomroes** [comment]
I tested the other two development streams. Both could start an LS that could connect via socket and simulate a model.

File: language-server/de.cau.cs.kieler.language.server/src/de/cau/cs/kieler/language/server/kicool/KeithCompilationUpdater.xtend [hunk 1]
```diff
@@ -38,7 +38,7 @@ class KeithCompilationUpdater implements Observer {
         }
     }
     
-    val KiCoolLanguageServerExtension kicoolExt
+    val de.cau.cs.kieler.language.server.kicool.KiCoolLanguageServerExtension kicoolExt
```
Comment (NiklasRentzCAU) [inline language-server/de.cau.cs.kieler.language.server/src/de/cau/cs/kieler/language/server/kicool/KeithCompilationUpdater.xtend:41]: Are these fully qualified IDs necessary here now? (appears in multiple files in the LS now)

**Review by NiklasRentzCAU — APPROVED** [review]
Found some minor things, LGTM otherwise. Works together with KLighD as well in the new setup and seems ready to be merged.

File: build/de.cau.cs.kieler.semantics.repository/category.xml [hunk 2]
```diff
@@ -32,14 +32,13 @@
...
-  <repository-reference location="http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.28.0/" enabled="true"/>
-  <repository-reference location="http://download.eclipse.org/lsp4j/updates/releases/0.15.0/" enabled="true"/>
-  <repository-reference location="http://kieler.github.io/KLighD/v2.2.0/" enabled="true"/>
-  <repository-reference location="http://rtsys.informatik.uni-kiel.de/~kieler/updatesite/release_pragmatics_2022-07" enabled="true"/>
+  <repository-reference location="https://download.eclipse.org/releases/2022-09" enabled="true"/>
+  <repository-reference location="https://download.eclipse.org/tools/orbit/downloads/drops/R20220830213456/repository/" enabled="true"/>
+  <repository-reference location="https://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.28.0/" enabled="true"/>
+  <repository-reference location="https://download.eclipse.org/lsp4j/updates/releases/0.15.0/" enabled="true"/>
+  <repository-reference location="https://kieler.github.io/KLighD/v2.3.0/" enabled="true"/>
+  <repository-reference location="https://rtsys.informatik.uni-kiel.de/~kieler/updatesite/release_pragmatics_2023-06" enabled="true"/>
```
[the remark this answers is not in the document]
Comment (a-sr) [reply build/de.cau.cs.kieler.semantics.repository/category.xml]: Fixed

**Pull request merged**